lib/linguist/heuristics.rb

```diff
@@ -68,6 +68,20 @@ def call(data)
     # Common heuristics
     ObjectiveCRegex = /^\s*(@(interface|class|protocol|property|end|synchronised|selector|implementation)\b|#import\s+.+\.h[">])/
 
+
+    CPPRegexType0 = /^\s*#.*include.*<(stdint|cstdint|string|vector|map|list|array|bitset|queue|stack|forward_list|unordered_map|unordered_set|(i|o|io)stream).*>/
```

Comment:
@rafael-santiago Try this:

~~~rb
CPPRegex = /
  ^ \s*
  (?: template \s* <
    | \#.*include.*<
      (stdint|cstdint|string|vector|map|list
      |array|bitset|queue|stack|forward_list
      |unordered_(map|set)|(i|o|io)stream).*>
    )
  |
  ^ [\x20\t]*
  (?: try \b
    | catch \s *\(
    | ( class
      | namespace
      | using \h+ namespace
      )
      \s+ \w+
    | (private|public|protected) :$
    )
  | std::\w+
  | .*\#ifdef.*__cplusplus.*
  | ^struct\s+
/xi
~~~

For regex with lots of nested capturing groups, it's much easier to maintain and review when they're indented.

When writing expressions in "expanded form" this way, be sure to escape all whitespace and `#` characters you're matching! Tabs and spaces are ignored, so they need to be written like `\x20` (space) and `\t` (tab). `#` characters are used for comments *inside* the regular expression: that's what I meant earlier before.

I don't think comments are necessary in this case: in fact, they're actually making things *harder* to follow. Stuff like URLs and sources are better off included in pull requests. ;)